includes/class-wc-stripe-status.php

```diff
@@ -194,4 +195,62 @@ public function render_status_report_section() {
 		</table>
 		<?php
 	}
+
+	/**
+	 * Add Stripe tools to the Woo debug tools.
+	 *
+	 * @param array $tools List of current available tools.
+	 */
+	public function debug_tools( $tools ) {
+		if ( WC_Stripe_Subscriptions_Helper::is_subscriptions_enabled() ) {
+			$tools['wc_stripe_list_detached_subscriptions'] = [
+				'name'     => __( 'List Stripe subscriptions with detached payment method', 'woocommerce-gateway-stripe' ),
+				'button'   => __( 'List subscriptions', 'woocommerce-gateway-stripe' ),
+				'desc'     => sprintf(
+					'%1$s<br/><strong class="red">%2$s</strong> %3$s<br/><strong>%4$s</strong>',
+					__( 'This tool will list all Stripe subscriptions with detached payment methods.', 'woocommerce-gateway-stripe' ),
+					__( 'Note:', 'woocommerce-gateway-stripe' ),
+					__( 'This tool will make an API request to Stripe for each active Stripe subscription in your store that is due to renew in the next month. For stores with many subscriptions, this may temporarily impact performance.', 'woocommerce-gateway-stripe' ),
+					__( 'Not recommended if you have more than 100 active subscriptions due for renewal within 30 days.', 'woocommerce-gateway-stripe' ),
+				),
+
+				'callback' => [ $this, 'list_detached_subscriptions' ],
+			];
+		}
+		return $tools;
+	}
+
+	/**
+	 * Lists Stripe subscriptions with detached payment methods.
+	 *
+	 * @return void
+	 */
+	public function list_detached_subscriptions() {
+		$subscriptions     = WC_Stripe_Subscriptions_Helper::get_detached_subscriptions();
```

Comment:
As a safety measure, might it be worth passing in a number like `1000` here rather than defaulting to `-1`?
```suggestion
		$subscriptions     = WC_Stripe_Subscriptions_Helper::get_detached_subscriptions( 1000 );
```

Reply:
And maybe we can add a notice (or an option to show a notice?) when a limit is supplied and additional subscriptions exist beyond that number? My only concern on this front is that we might not have that iteration in place before we ship 9.6.0 next week.

Reply:
The problem I see with that is that we would need to query for all subscriptions to check how many might exist beyond 1000.

> My only concern on this front is that we might not have that iteration in place before we ship 9.6.0 next week.

It was not my intention to target 9.6.0. As I mentioned during the meetup, it was preferred that 9.6.0 focus on legacy 👍 . The idea was to ship this before or after it.